Further clarification about establishing trust with RP and CI (#453)
italia/eidas-it-wallet-docs: docs/en/trust.rst

    The Wallet Attestation conveys all the required information pertaining to the instance, such as its public key and any other technical or administrative information, without any User's personal data.

> Establishing Trust with Relying Parties

Reviewer: There is a duplicated section below

Author: o, yes.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The Relying Party is registered by a Trust Anchor or its Intermediate and obtains a Trust Mark to be included in its Entity Configuration.
In its Entity Configuration the Relying Party publishes its specific metadata, including the supported signature and encryption algorithms
and any other necessary information for the interoperability requirements.

> Any requests for User attributes, such as PID or (Q)EAA, from the Relying Party to Wallet Instances are signed and SHOULD contain the verifiable

Reviewer: ```suggestion
Any requests for User attributes, such as PID or (Q)EAA, from the Relying Party to Wallet Instances are signed using one of the protocol related keys included in the metadata of the Relying Party.
```

Author: according to the work made so far in openid federation wallet arch draft, the cryptographic material will be definitively provided by .well-known/jwt-vc-issuer and this will be reflected also in openid_credential_issuer

since these changes are not supposed to be acquired in this release, I suggest to keep it generic for now and reduce the amount of changes we expect for the next releases

Reviewer: I don't understand your comment. In the suggestion I referred to openid_relying_party metadata and not metadata of credential Issuers.

Author: unfortunately this applyies to both credential issuer and verifiers, since them both are jwt vc issuer

since this is a huge conversation still on the way, I suggest to relax our wording making it more flexible and applicable with different technical solutions

the end of this road will happen with the evolution of the openid4vci specs and the federation wallet arch (in course of approval)

for the time being, I would reduce the amount of fixed stuffs in multiple parts, to reduce the amount of changes we might have in the future

Trust Chain regarding the Relying Party.

The Wallet Instance verifies that the Trust Chain related to the Relying Party is still active,
proving that the Relying Party is still part of the Federation and not revoked.

> The Trust Chain MAY be contained within the signed request in the form of a JWS header parameter.

Reviewer: ```suggestion
The Trust Chain MAY be contained within the signed request in the form of a JWS header parameter, or dynamically built though a Federation Entity Discovery process.
```

> The Trust Chain MAY be contained within the signed request in the form of a JWS header parameter, or dynamically built though a Federation Entity Discovery process.

Author: ```suggestion
The Trust Chain MAY be contained within the signed request in the form of a JWS header parameter, or dynamically built through a Federation Entity Discovery process.
```

Establishing Trust with Credential Issuers
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

In the issuance process, trust evaluation ensures the integrity and authenticity of the Credentials being issued and the realiability of their Issuers. This section delineates the trust evaluation mechanisms distinct from the protocol flows, implemented by Wallet Instances and Relying Parties, as described in the dedicated section.

Trust evaluations implement different ways, as defined below:

* **Federation Entity Discovery**: Wallet Instances and Relying Parties MUST verify the identity of the Issuer through a Federation Entity Discovery process. This involves querying a trusted list or directory to confirm the Issuer's validity status and compliance with the Trust Framework.

* **Trust Chains**: Wallet Instances and Relying Parties evaluate Issuer's Trust Chains, be provided statically or build though a Federation Entity Discovery process, to ensure that the entity requesting the Credential is part of a recognized and trusted federation. This involves checking the Trust Chain from the root authority to the Issuer.

> * **Trust Marks Evaluation**: Trust Marks are assessed to ensure ongoing compliance with federation policies. These marks indicate adherence to specific standards and practices required by the federation.

Reviewer: Trust marks are currently not required by our Trust model. We should introduce them as required and add more technical details about the usage of trust marks. Maybe we can cover this aspect in a future release.

Author: trust models is all about high level mechanisms, including entities involved such the trusted third party.

I suggest to keep the current approach describing how to issues policy, looking forward for the definitions of the policies we want to implement in it-wallet

Reviewer: I agree that we will define policies and trus marks in a future release but, according to this, we should remove these steps introducing them when we will have policies and trust marks. Otherwise this section will be unclear. The aim of this section is to provide clear instructions for developers to implement trust establishment mechanisms.

Author: introducing features like trust marks here, enable the definition of these trust marks

according to the evidence that we will not have the presentation_definitions_supported within the RP metadata, we definitively move out policies issuance and evaluation using trust marks

therefore I would leave them here and work in 0.9.0 on the definition of these

> * **Policy Evaluation**: Wallet Instances and Relying Parties MUST check that the Credential Issuer is allowed in the issuance of the Credential of their interest. Metadata, metadata policies and Trust Marks are used for the implementation of these checks.

Reviewer: Same comment as for the trust marks applies here.

Author: as above
